lib/ui/painting.dart

```diff
@@ -1873,7 +1872,10 @@ class _Image extends NativeFieldWrapperClass1 {
     _dispose();
   }
 
-  void _dispose() native 'Image_dispose';
+  /// This can't be a leaf call because the native function calls Dart API
+  /// (Dart_SetNativeInstanceField).
```

Comment:
In that case we'd get a crash.
The leaf call is faster because it doesn't bother with all the usual bookkeeping of frames, VM state transitions, and register setup. So one of the first issues we're likely to encounter is some of the VM sanity checks failing an assert.
Alternatively if anything runs that relies on those usual assumptions, we'd get a segfault or similar as the stack corrupts.

Unfortunately we'd need some pretty sophisticated analysis to automatically determine whether a native function is "leaf safe".